src/main/java/org/opensearch/knn/index/Parameter.java

```diff
@@ -63,10 +64,19 @@ public IntegerParameter(Integer defaultValue, Predicate<Integer> validator)
         }
 
         @Override
-        public void validate(Object value) {
-            if (!(value instanceof Integer) || !validator.test((Integer) value)) {
-                throw new ValidationException();
+        public ValidationException validate(Object value) {
+            ValidationException validationException = null;
+            if (!(value instanceof Integer)) {
+                validationException = new ValidationException();
+                validationException.addValidationError("Value not of type Integer for Integer parameter.");
+                return validationException;
```

Comment:
Is it possible to add which parameter value is not integer?

Reply:
I didnt include parameter name as part of parameter class. I can probably add this.